hasher.go

```diff
@@ -284,6 +285,82 @@ func (h *Hasher) MerkleizeWithMixin(indx int, num, limit uint64) {
 	h.buf = append(h.buf[:indx], input...)
 }
 
+// MerkleizeGoHashTree merkleizes hasher's buffered data using the gohashtree library.
+func (h *Hasher) MerkleizeGoHashTree(ix int) {
+	inputLen := len(h.buf[ix:])
+	if inputLen == 0 {
+		h.buf = append(h.buf[:ix], zeroBytes...)
+		return
+	}
+
+	// calculate the nearest power of 2 not smaller than inputLen
+	twoToPower := 1
+	for twoToPower < inputLen {
+		twoToPower *= 2
+	}
+
+	result, err := h.calculateHash(ix, uint64(twoToPower))
+	if err != nil {
+		panic(err)
```

Comment:
The panic is temporary. I will change the function's signature once we introduce toggling between this algorithm and the old one.